rocAL/include/decoders/audio/audio_decoder.h

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+#ifdef ROCAL_AUDIO
+#include "sndfile.h"
+
+class AudioDecoder {
+   public:
+    enum class Status {
+        OK = 0,
+        HEADER_DECODE_FAILED,
+        CONTENT_DECODE_FAILED,
+        UNSUPPORTED,
+        FAILED,
+        NO_MEMORY
+    };
+    virtual AudioDecoder::Status initialize(const char* src_filename) = 0;                           // This function is responsible for initializing the audio decoder. It takes the source filename as input and returns the status of the initialization process.
```

Comment:
All class function names should start with capital letters. "Initialize". Follow [Google coding standards](https://google.github.io/styleguide/cppguide.html) for all files in the PR.

Reply:
Add function comments above the function in Doxygen supported format across all files